### PR TITLE
Create a new enode and generate a dv5 service

### PR DESCRIPTION
**Pull request by Ryan0v0**
## Motivation
This PR aims to generate the first draft of the `eth-light-crawler` using discovery5.

## List of Tasks TODO:
- [x] Create a new `enode` with a new `PrivKey`
- [x] generate a `dv5 service` with that particular `enode`
- [x] start reading random `ENRs` from the network
- [ ] parse the received `ENRs`
- [ ] persist the `ENRs` somewhere (local storage, PSQL)

### CODE REVIEW

**Review by cortze — COMMENTED** [review]
It would be nice if you could keep the `.idea/` folder from `XCode` outside the repo. You can add it to the `.gitignore`

File: cmd/discovery5.go [hunk 1]
```diff
@@ -80,6 +83,28 @@ func GenNewPrivKey() (*ecdsa.PrivateKey, error) {
 	return ecdsa.GenerateKey(gcrypto.S256(), rand.Reader)
 }
 
+var EthBootonodes = []*enode.Node{
```
Comment (cortze) [inline cmd/discovery5.go:86]: Hey @Ryan0v0 , I don't see the point of duplicating the Bootnodes. 
They were already in [pkg/config/bootnodes.go](https://github.com/migalabs/eth-light-crawler/blob/main/pkg/config/bootnodes.go)

File: cmd/discovery5.go [hunk 2]
```diff
@@ -103,5 +128,15 @@ func RunDiscv5(ctx *cli.Context) error {
 	// Discovery5 service
 	// check: https://github.com/migalabs/armiarma/blob/ca3d2f6adea364fc7f38bdabda912b5541bb4154/src/discovery/dv5/dv5_service.go#L58
 	// Bootnodes are in pkg/config/bootnodes
```
Comment (cortze) [inline cmd/discovery5.go:130]: Coming from the previous [comment](https://github.com/migalabs/eth-light-crawler/pull/1/commits/8db8a34c3974439e6c107152a416d2993b25c686#r1045785021) , you could directly ref the Bootnodes from `pkg/config/bootnodes`

File: cmd/discovery5.go [hunk 3]
```diff
@@ -41,14 +47,46 @@ type Eth2InfoData struct {
 	OutputPath    string
 }
 
+func GeneratePrivKey() string {
```
Comment (cortze) [inline cmd/discovery5.go:53]: I'm not sure why do we need two functions that Generate a `ecdsa.PrivateKey`?

As far as I know, the Ethereum Nodes only need the `*ecdsa.PrivateKey` to sign messages. Therefore:
```go
func GenNewPrivKey() (*ecdsa.PrivateKey, error) {
	return ecdsa.GenerateKey(gcrypto.S256(), rand.Reader)
```
should work

File: cmd/discovery5.go [hunk 4]
```diff
@@ -103,5 +128,15 @@ func RunDiscv5(ctx *cli.Context) error {
 	// Discovery5 service
 	// check: https://github.com/migalabs/armiarma/blob/ca3d2f6adea364fc7f38bdabda912b5541bb4154/src/discovery/dv5/dv5_service.go#L58
 	// Bootnodes are in pkg/config/bootnodes
+	c := cli.Context
+	infObj, _ := info.InitEth2(c)
+	dv5, _ := dv5.NewDiscovery(
+		ctx.Context,
+		ln, // LocalNode?
+		(*ecdsa.PrivateKey)(infObj.PrivateKey),
```
Comment (cortze) [inline cmd/discovery5.go:136]: Seems that we don't need such a complex casting: 
you could just use the `*ecdsa.Privatekey` from the function `GenNewPrivKey()`

**Review by cortze — CHANGES_REQUESTED** [review]
Nice progress, left you some comments that will help you keep the code clean :)

File: cmd/discovery5.go (hunk 2 above)
Comment (Ryan0v0) [reply cmd/discovery5.go:130]: How do I call the Bootnodes in `pkg/config/bootnodes`?

Comment (cortze) [reply cmd/discovery5.go:130]: You can refer to the `config` package in the import section on top of it 
```go
import(
     ...
     "github.com/migalabs/eth-light-crawler/pkg/config"
     ...
)


...
bootnodes := config.Bootnodes 
...
``

**Comment by cortze** [comment]
Closing this PR due to #3

**Pull request closed without merging**